### ISSUE TITLE
3.22.1 compile problems on FreeBSD

### ISSUE THREAD

**Issue by czanik** [balabit/syslog-ng#2805]
# syslog-ng

## Version of syslog-ng

3.22.1 release

## Platform

root@fb120rc1:~/syslog-ng322 # uname -a
FreeBSD fb120 12.0-RELEASE-p6 FreeBSD 12.0-RELEASE-p6 GENERIC  amd64

# Issue

## Failure

When 3.22.1 release is compiled with default options (ipv6, json, http) on FreeBSD, it compiles fine. Enabling most of the additional features also work, but not all. I still have to pinpoint all of the failing modules. One example is MaxMinDB support. 

```
libtool: link: cc -Wshadow -fcommon -Wno-stack-protector -Wno-unused-parameter -Wno-variadic-macros -O2 -g -pthread -Wno-initializer-overrides -O2 -pipe -fstack-protector-strong -fno-strict-aliasing -fstack-protector-strong -o syslog-ng/.libs/syslog-ng syslog-ng/main.o -Wl,--export-dynamic -pthread  -L/usr/local/lib -L./lib/.libs -lsyslog-ng -lrt -lgmodule-2.0 -lgthread-2.0 -lglib-2.0 -lintl ./lib/eventlog/src/.libs/libevtlog.so ./lib/secret-storage/.libs/libsecret-storage.so -lpcre -lregex -ldl -pthread -Wl,-rpath -Wl,/usr/local/lib
/usr/bin/ld: error: undefined symbol: timeutils_setup_timezone_hook
>>> referenced by main.c:281 (syslog-ng/main.c:281)
>>>               syslog-ng/main.o:(main)
cc: error: linker command failed with exit code 1 (use -v to see invocation)
gmake[4]: *** [Makefile:12626: syslog-ng/syslog-ng] Error 1
gmake[3]: *** [Makefile:20981: all-recursive] Error 1
gmake[2]: *** [Makefile:8638: all] Error 2
gmake[2]: Leaving directory '/root/syslog-ng322/work/syslog-ng-3.22.1'
*** Error code 1
```

**Comment by czanik** [comment]
Just finished testing: the other affected feature is the SQL destination.

Oh, and I did not test the SNMP destination yet, as it is not yet added to the port.

**Comment by czanik** [comment]
OK. Fixed packaging list for debuggercli and added SNMP destination support. It compiles without any problems, so the two affected modules are:
 - geoip2
 - sql

**Comment by MrAnno** [comment]
Do you have an older syslog-ng version installed on this machine?

`syslog-ng/main` is linked to libsyslog-ng using `-lsyslog-ng` instead of using the full path (or via libtool) for some reason (see below), so it is possible it might try to link to an older installed version, where `timeutils_setup_timezone_hook()` does not exist.

https://github.com/balabit/syslog-ng/blob/67f9f32dc38cba0e970e37d573b249fca81b701c/syslog-ng/Makefile.am#L12-L17

**Comment by czanik** [comment]
Yes, syslog-ng 3.21 was installed. Once I removed it, compilation worked all options enabled.

I leave this problem open, as while larger installations are usually not affected (they often have dedicated package building machines starting builds in a clean environment) for standalone installations the workflow is, that one builds the new package, and once it is successfully built, the old package is only removed then.

**Comment by MrAnno** [comment]
Okay, we'll fix it. Thanks!

**Comment by lbudai** [comment]
@czanik : could you try it with cmake?
(and I think we should close this issue...)

**Comment by czanik** [comment]
Just tested it again with still the regular ```make``` command and it still fails.

This week is a kind of short for me (on holiday most of the week), but next Monday or Thursday I'd like to sit down with someone in the developers team and try ```cmake``` together. Last time we checked we ran into multiple problems.

Once it works outside of ```FreeBSD ports``` I will turn the syslog-ng ports to use cmake.
